integrations/tensorflow/e2e/keras/layers/layers_test.py

```diff
@@ -16,528 +16,463 @@
 
 import collections
 import copy
+import inspect
 import os
-from typing import Any, Dict, Sequence, Union
+from typing import Any, Dict, List, Sequence, Tuple, Union
 
 from absl import app
 from absl import flags
+from absl import logging
 from pyiree.tf.support import tf_test_utils
 from pyiree.tf.support import tf_utils
 import tensorflow.compat.v2 as tf
 
 FLAGS = flags.FLAGS
 
 DROPOUT = 0.5
-DIM = 4
-RANK_2_INPUT = [DIM] * 2
-RANK_3_INPUT = [DIM] * 3
-RANK_4_INPUT = [DIM] * 4
-
-CONV_1D_INPUT = [2, 8, 3]
-CONV_2D_INPUT = [2, 8, 8, 3]
-CONV_3D_INPUT = [2, 8, 8, 8, 3]
-
-# Configs are namedtuples storing keyword arguments and shapes to test a
-# tf.keras.layers.Layer with. They are used in two ways:
-#   1. To directly specify the kwargs and shapes for a layers test.
-#   2. In 'generate_configs', to specify how to change a default config to
-#      specify a non-default test. In this case, the overriding Config will
-#      exclusively specify the shape of the test if its shape is not None, and
-#      the overriding Config will extend/update the kwargs of the default
-#      Config.
-Config = collections.namedtuple('Config', ['kwargs', 'shapes'])
-# Use old default API for compatibility with Python 3.6.
-Config.__new__.__defaults__ = (dict(), None)
-
-
-def generate_configs(default_config: Config,
-                     override_configs: Dict[str, Config]) -> Dict[str, Config]:
-  """Generates a dict of 'Config's based off changes to a default Config."""
-  configs = {'default': default_config}
-  for exported_name, config in override_configs.items():
-    shapes = default_config.shapes if config.shapes is None else config.shapes
-
-    # Deep copy to avoid inplace mutation of the default.
-    kwargs = copy.deepcopy(default_config.kwargs)
-    kwargs.update(config.kwargs)  # Adds new and overwrites old kwargs.
-
-    configs[exported_name] = Config(kwargs, shapes)
-  return configs
-
-
-# A dict mapping tf.keras.layers names to either a single Config (representing
-# the kwargs and shapes to use to test a Layer) or a dict mapping exported_names
-# to Configs. The latter case is usually automatically generated via
-# 'generate_configs', with the 'Config's in 'override_configs' specifying how
-# to modify the 'default_config's kwargs and shapes.
+CONV_FILTERS = 2
+CONV_KERNEL_SIZE = 2
+DIM = 3
+
+# Used for attention layers and recurrent layers.
+RANK_3_SHAPE = [DIM] * 3
+# Highest rank that tf.keras will allow for all layers.
+RANK_5_SHAPE = [DIM] * 5
+
+UNARY_SIGNATURE_SHAPES = [[RANK_5_SHAPE]]
+BINARY_SIGNATURE_SHAPES = [[RANK_5_SHAPE] * 2]
+TERNARY_SIGNATURE_SHAPES = [[RANK_5_SHAPE] * 3]
+
+CONV_1D_SIGNATURE_SHAPES = [[[2, 8, 3]]]
+CONV_2D_SIGNATURE_SHAPES = [[[2, 8, 8, 3]]]
+CONV_3D_SIGNATURE_SHAPES = [[[2, 8, 8, 8, 3]]]
+
+RNN_SIGNATURE_SHAPES = [[RANK_3_SHAPE]]
+RNN_KWARGS_TO_VALUES = dict(units=[4],
```

Comment:
I did it unconsciously, but I think it's reasonable to assume that it was because it makes the `kwargs_to_values` variables visually distinct from the `LAYERS_TO_*` variables. I tried converting the latter, and the effect is pretty difficult to read:
<img width="551" alt="Screen Shot 2020-11-20 at 4 37 31 PM" src="https://user-images.githubusercontent.com/40036383/99862719-9f4d8300-2b4f-11eb-8746-303e8667db92.png">